src/Illuminate/Database/Eloquent/Relations/Concerns/AsPivot.php

```diff
@@ -110,11 +114,22 @@ protected function setKeysForSaveQuery(Builder $query)
      */
     public function delete()
     {
+        // support for pivot classes that container a non-composite primary key
         if (isset($this->attributes[$this->getKeyName()])) {
-            return parent::delete();
+            return (int) parent::delete();
         }
 
-        return $this->getDeleteQuery()->delete();
+        if ($this->fireModelEvent('deleting') === false) {
+            return 0;
+        }
+
+        $this->touchOwners();
+
+        $this->getDeleteQuery()->delete();
+
+        $this->fireModelEvent('deleted', false);
+
+        return 1;
```

Comment:
Shouldn't still be the result of `$this->getDeleteQuery()->delete();` ?

Reply:
Yeah, that would be more accurate.